assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -1975,6 +1979,17 @@ private String build_view_external_tool_launch_context(VelocityPortlet portlet,
                 context.put("value_ContentLaunchURL", content_launch);
                 context.put("placement", "assignment_launch_"+contentKey);
 
+                Long toolKey = new Long(content.get(LTIService.LTI_TOOL_ID).toString());
```

Comment:
new Long() is discouraged for performance reasons and you should use Long.valueOf()

Reply:
All set.  This number will soon be > 128 so the performance benefit will be limited :)  But it is the new practice.